types/hubot/index.d.ts

```diff
@@ -1,10 +1,12 @@
+/// <reference types="node" />
 import { EventEmitter } from "events";
 import { Express } from "express";
 import { Server } from "http";
 import { Options as HttpOptions, ScopedClient } from "scoped-http-client";
 
 declare namespace Hubot {
     class Adapter extends EventEmitter {
+        robot: Robot;
```

Comment:
without this line robot isn't able to be referenced from incoming messages

https://github.com/hubotio/hubot/blob/main/src/Adapter.mjs#L11